pkg/azure/ipam/instances.go

```diff
@@ -46,6 +46,14 @@ func (m *InstancesManager) CreateNode(obj *v2.CiliumNode, n *ipam.Node) ipam.Nod
 	return &Node{manager: m, node: n}
 }
 
+// HasInstance returns whether the instance is in instances
+func (m *InstancesManager) HasInstance(instanceID string) bool {
+	m.mutex.RLock()
+	instanceExist := m.instances.Exists(instanceID)
```

Comment:
nit: might be a bit cleaner to just return the result of .Exists() and defer the RUnlock()